pkgs/applications/office/p3x-onenote/default.nix

```
@@ -0,0 +1,57 @@
+{ stdenv, appimageTools, autoPatchelfHook, desktop-file-utils
+  , fetchurl, runtimeShell }:
+
+let
+  version = "2020.10.111";
+  name = "p3x-onenote-${version}";
+
+  plat = {
+    aarch64-linux = "-arm64";
+    armv7l-linux = "-armv7l";
+    i386-linux = "-i386";
+    i686-linux = "-i386";
+    x86_64-linux = "";
+  }.${stdenv.hostPlatform.system};
+
+  sha256 = {
+    aarch64-linux = "0a3c0w1312l6k2jvn7cn8priibnh8wg0184zjcli29f9ds1afl5s";
+    armv7l-linux = "172m2d94zzm8q61pvnjy01cl5fg11ad9hfh1han0gycnv3difniy";
+    i386-linux = "12m0i5sb15sbysp5fvhbj4k36950m7kpjr12n88r5fpkyh13ihsp";
+    i686-linux = "12m0i5sb15sbysp5fvhbj4k36950m7kpjr12n88r5fpkyh13ihsp";
+    x86_64-linux = "0bn48r55l5dh8zcf8ijh3z6hlyp3s6fvfyqc1csvnslm63dfkzcq";
+  }.${stdenv.hostPlatform.system};
+
+  src = fetchurl {
+    url = "https://github.com/patrikx3/onenote/releases/download/v${version}/P3X-OneNote-${version}${plat}.AppImage";
+    inherit sha256;
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook desktop-file-utils ];
+in
+appimageTools.wrapType2 rec {
+  inherit name src;
+
+  extraInstallCommands = ''
+    mkdir -p $out/share/pixmaps $out/share/licenses/${name}
+    cp ${appimageContents}/p3x-onenote.png $out/share/pixmaps/
+    cp ${appimageContents}/p3x-onenote.desktop $out
+    cp ${appimageContents}/LICENSE.electron.txt $out/share/licenses/${name}/LICENSE
```

would also be nice to have a more stable command name, currently it's of the shame `${pname}-${version}`

```
$ tree ./results/p3x-onenote/bin/
./results/p3x-onenote/bin/
└── p3x-onenote-2020.10.111
```

```suggestion
    cp ${appimageContents}/LICENSE.electron.txt $out/share/licenses/${name}/LICENSE
    ln $out/bin/${name} $out/bin/p3x-onenote
```

---

That's my bad, I didn't know it'd change the name of the binary to `name`. I honestly didn't expect it to really, that seems odd. Do you know the reasoning behind it?

In any case, we should probably make wrapType2 able to handle pname and version so that this issue can be avoided entirely.

---

it's most likely how the appimage tools works, and changing it at the wrapType level will probably affect many more packages.

---

rather than keeping both I've simply renamed it